Objects/codeobject.c

```diff
@@ -1906,3 +1906,18 @@ _PyCode_ConstantKey(PyObject *op)
     }
     return key;
 }
+
+void 
+_PyStaticCode_Dealloc(PyCodeObject *co, _Py_CODEUNIT *firstinstr)
```

Comment:
Why does `firstinstr` need to be passed in here? Won't it always be reset to point to the first data byte of `co->co_code`?

Reply:
It is fixed in a later, I did it because currently there are two copies of the code, one specialized, one non-specialized which wastes memory so if in future they will be combined then this will be required although it is currently now required.

Reply:
If you're hoping that we'll get to the point where the unspecialized code is no longer needed, the specialization architecture actually explicitly requires the unspecialized code to stick around. (I think PEP 659 might explain why, if you cannot find it, I can give you the rundown.)

Since this is a purely internal API there's little cost to changing it in the future, but there's a definite cost in this redundant design (fortunately Mark fixed it already).